pandas/core/apply.py

```diff
@@ -424,11 +377,19 @@ def apply_broadcast(self, target: "DataFrame") -> "DataFrame":
 
     @property
     def series_generator(self):
-        constructor = self.obj._constructor_sliced
-        return (
-            constructor(arr, index=self.columns, name=name)
-            for i, (arr, name) in enumerate(zip(self.values, self.index))
-        )
+        values = self.values
+        assert len(values) > 0
+
+        # We create one Series object, and will swap out the data inside
+        #  of it.  Kids: don't do this at home.
+        ser = self.obj._ixs(0, axis=0)
+        mgr = ser._mgr
+        blk = mgr.blocks[0]
+
+        for (arr, name) in zip(values, self.index):
```

Comment:
can you push this to an internals method instead?

Reply:
im looking at that now.  the other place where this pattern could be really useful is in groupby.ops, but its tougher there

Reply:
sure also exposing an api for this would be ok as well (eg another internals method)

Reply:
im still troubleshooting the groupby.ops usage, would like to punt on making this an internals method for the time being